dev/com.ibm.ws.ejbcontainer/src/com/ibm/ws/ejbcontainer/osgi/internal/EJBMDOrchestratorImpl.java

```diff
@@ -1342,32 +1344,43 @@ protected void processZOSMetadata(BeanMetaData bmd) {
         if (isTraceOn && tc.isEntryEnabled())
             Tr.entry(tc, "processZOSMetadata : " + bmd);
 
-        if (EJSPlatformHelper.isZOS()) {
-            bmd.m_syncToOSThreadValue = getSyncToOSThreadSetting(ContainerConfigConstants.syncToOSThreadSetting, bmd);
+        if (isZOS()) {
+            // Call the BeanMetaData's initializeSyncToOSThread method to check environment entry
+            bmd.initializeSyncToOSThread();
+            
+            // If not set by environment entry, check the binding
+            if (!bmd.m_syncToOSThreadValue) {
+                bmd.m_syncToOSThreadValue = getSyncToOSThreadSetting(ContainerConfigConstants.syncToOSThreadSetting, bmd);
+            }
+            
             if (bmd.m_syncToOSThreadValue) {
                 if (isTraceOn && tc.isDebugEnabled())
-                    Tr.debug(tc, "From the bean's env-var, m_syncToOSThreadValue is set to true");
-
-                // check to if server is enabled for SyncToOSThread
-                // WSLoginLocalOSExtension localSecurityLoginExt;
-                // localSecurityLoginExt = com.ibm.ws.security.auth.j2c.WSLoginLocalOSExtensionFactory.getInstance();
-                // if (!(localSecurityLoginExt.isApplicationSyncToOSThreadEnabled())) {
-                //     Tr.warning(tcWS390, "BBOJ0081", bmd.j2eeName);
-                //     bmd.m_syncToOSThreadValue = false;
-                // } else {
-                //     if (isTraceOn && tc.isDebugEnabled())
-                //         Tr.debug(tc, "EJB requests SynToOSThread, and the server is enabled for SyncToOSThread");
-                // }
+                    Tr.debug(tc, "SyncToOSThread is enabled for bean: " + bmd.enterpriseBeanName);
+
+                // Check if ThreadIdentityManager supports application thread identity
+                if (ThreadIdentityManager.isAppThreadIdentityEnabled()) {
```

Comment:
It appears this should be the only part of this method that is updated.

Reply:
Oh, and since the call to ThreadIdentityManager is only being performed for trace, it should be inside the trace guarding... like:
```
if (isTraceOn && tc.isDebugEnabled()) {
    if (ThreadIdentityManager.isAppThreadIdentityEnabled()) {
        Tr.debug(tc, "EJB requests SyncToOSThread, and the server is enabled for SyncToOSThread");
    } else {
        Tr.debug(tc, "EJB requests SyncToOSThread, but the server does not have ThreadIdentityService enabled");
        // Keep the setting enabled - the ThreadIdentityManager will handle the case when no service is available
    }
}
```
Unless there is some required side effect to calling ThreadIdentityManager, it would be better to not call it unless trace is enabeld.